chain/client/src/client.rs

```diff
@@ -443,10 +448,21 @@ impl Client {
                 ) {
                     // By now the chunk must be in store, otherwise the block would have been orphaned
                     let chunk = self.chain.get_chunk(&chunk_header.chunk_hash()).unwrap();
+
+                    let validated_txs = ValidatedTransaction::new_list(
+                        &config,
+                        chunk.transactions().into_iter().cloned(),
+                    )
+                    .map_err(|(err, signed_tx)| {
+                        Error::Other(format!(
+                            "Validating signed tx ({:?}) failed with {:?}",
+                            signed_tx, err
+                        ))
+                    })?;
```

Comment:
I'm suspicious of whether we should early return here or continue processing transactions, skipping the invalid ones. At least from my quick reading `reintroduce_transactions` wouldn't fail if there were invalid transactions before and chunk production would only discard individual invalid transactions, not entire groups or something.

If this retains the existing behaviour, though, then fine, but can you please walk me through where the current behaviour matches this change?

Reply:
Otherwise LGTM.

Reply:
Good catch, you are right.  I think this is a copy pasta error.  I'll fix.

Reply:
now fixed.